core/src/main/java/rdx/works/core/sargon/serializers/SargonSignatureSerializer.kt

```diff
@@ -0,0 +1,123 @@
+package rdx.works.core.sargon.serializers
+
+import com.radixdlt.sargon.PublicKey
+import com.radixdlt.sargon.Signature
+import com.radixdlt.sargon.SignatureWithPublicKey
+import com.radixdlt.sargon.extensions.asGeneral
+import com.radixdlt.sargon.extensions.hex
+import com.radixdlt.sargon.extensions.hexToBagOfBytes
+import com.radixdlt.sargon.extensions.init
+import com.radixdlt.sargon.extensions.string
+import kotlinx.serialization.KSerializer
+import kotlinx.serialization.SerializationException
+import kotlinx.serialization.descriptors.PrimitiveKind
+import kotlinx.serialization.descriptors.PrimitiveSerialDescriptor
+import kotlinx.serialization.descriptors.SerialDescriptor
+import kotlinx.serialization.descriptors.buildClassSerialDescriptor
+import kotlinx.serialization.descriptors.element
+import kotlinx.serialization.encoding.CompositeDecoder
+import kotlinx.serialization.encoding.Decoder
+import kotlinx.serialization.encoding.Encoder
+import kotlinx.serialization.encoding.decodeStructure
+import kotlinx.serialization.encoding.encodeStructure
+
+class SargonSignatureSerializer : KSerializer<Signature> {
+
+    override val descriptor: SerialDescriptor =
+        PrimitiveSerialDescriptor("com.radixdlt.sargon.Signature", PrimitiveKind.STRING)
+
+    override fun serialize(encoder: Encoder, value: Signature) {
+        encoder.encodeString(value.string)
+    }
+
+    override fun deserialize(decoder: Decoder): Signature {
+        return Signature.init(decoder.decodeString().hexToBagOfBytes())
+    }
+}
+
+class SargonPublicKeySerializer : KSerializer<PublicKey> {
+
+    override val descriptor: SerialDescriptor =
+        PrimitiveSerialDescriptor("com.radixdlt.sargon.PublicKey", PrimitiveKind.STRING)
+
+    override fun serialize(encoder: Encoder, value: PublicKey) {
+        encoder.encodeString(value.hex)
+    }
+
+    override fun deserialize(decoder: Decoder): PublicKey {
+        return PublicKey.init(decoder.decodeString())
+    }
+}
+
+@Suppress("CyclomaticComplexMethod")
+class SignatureWithPublicKeySerializer : KSerializer<SignatureWithPublicKey> {
+
+    override val descriptor: SerialDescriptor =
+        buildClassSerialDescriptor("com.radixdlt.sargon.SignatureWithPublicKey") {
+            element<String>("type")
+            element("publicKey", buildClassSerialDescriptor("publicKey"))
+            element("signature", buildClassSerialDescriptor("signature"))
+        }
+
+    override fun serialize(encoder: Encoder, value: SignatureWithPublicKey) {
+        encoder.encodeStructure(descriptor) {
+            when (value) {
+                is SignatureWithPublicKey.Secp256k1 -> {
+                    encodeStringElement(descriptor, 0, "Secp256k1")
```

Comment:
I would suggest putting those strings into a private constant since you reuse them a lot.